src/main/java/com/jcabi/log/Logger.java

```diff
@@ -368,6 +387,14 @@ public void write(final int data) throws IOException {
         } else {
             logger = LoggerFactory.getLogger(source.getClass());
         }
+        try {
+            final Field fqcn = logger.getClass()
+                .getDeclaredField("FQCN");
+            setFinalStatic(fqcn, Logger.class.getName());
+            // @checkstyle EmptyBlockCheck (5 lines)
+        } catch (final NoSuchFieldException ex) {
+        } catch (final IllegalAccessException ex) {
+        }
```

Comment:
@pecko this is really weird... I don't think I can accept that... if you really want to leave it like this, you should put a proper explanation of this solution, into javadoc of this method